From 3aceba33a5c4c1cfff8ef0ae3f34db123e9fe3e2 Mon Sep 17 00:00:00 2001
From: Dan Williams <dan.j.williams@intel.com>
Date: Thu, 10 Mar 2016 07:48:35 -0800
Subject: [PATCH] ndctl: add '--disable-docs' option to configure

For environments that want to build without the documentation add an
option to omit the Documentation/ sub-directory.

Reported-by: Arjan van de Ven <arjan@linux.intel.com>
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
---
 Makefile.am  | 2 ++
 configure.ac | 9 +++++++++
 2 files changed, 11 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index ce97a15..8458277 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,7 @@
 EXTRA_DIST =
+if ENABLE_DOCS
 SUBDIRS = . Documentation
+endif
 CLEANFILES =
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 AM_MAKEFLAGS = --no-print-directory
diff --git a/configure.ac b/configure.ac
index 9506e8e..96a02d9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,6 +58,15 @@ AS_IF([test "x$enable_logging" = "xyes"], [
         AC_DEFINE(ENABLE_LOGGING, [1], [System logging.])
 ])
 
+AC_ARG_ENABLE([docs],
+        AS_HELP_STRING([--disable-docs],
+	[disable documentation build @<:@default=enabled@:>@]),
+        [], enable_docs=yes)
+AS_IF([test "x$enable_docs" = "xyes"], [
+        AC_DEFINE(ENABLE_DOCS, [1], [Documentation / man pages.])
+])
+AM_CONDITIONAL([ENABLE_DOCS], [test "x$enable_docs" = "xyes"])
+
 AC_ARG_ENABLE([debug],
         AS_HELP_STRING([--enable-debug], [enable debug messages @<:@default=disabled@:>@]),
         [], [enable_debug=no])
-- 
2.8.2

